packages/libwasmer/libwasmer.3.0.0~beta/files/libwasmer.install

```diff
@@ -0,0 +1,3 @@
+lib: [
+	"package/lib/libwasmer.so"
```

Comment:
Ah, nevermind there seems to be an issue on macOS where this file doesn't exist. Shouldn't it be `libwasmer.dylib` there? Or maybe you'd prefer to take the static archive?

Reply:
if you really need the dynamic library over the static one:
```suggestion
	"?package/lib/libwasmer.so"
	"?package/lib/libwasmer.dylib"
	"?package/lib/libwasmer.dll"
```

Reply:
If your package works with the static archive only the following should be necessary:
```suggestion
	"package/lib/libwasmer.a"
```

Reply:
It should be, but this file is OS-independent and Dune **only** supports the `.so` extension on MacOS. So I need to rename the library after building it. (Also the package doesn't work or generate massive binaries with the static library, so I need to use the dynamic one.)
I've added the Windows extension for future-proofing though.

Reply:
Actually, according to the Wasmer Makefile:
```
if [ -f target/release/wasmer.dll ]; then \
	cp target/release/wasmer.dll package/lib/wasmer.dll ;\
fi
if [ -f target/release/wasmer.dll.lib ]; then \
	cp target/release/wasmer.dll.lib package/lib/wasmer.dll.lib ;\
fi
if [ -f target/release/wasmer.lib ]; then \
	cp target/release/wasmer.lib package/lib/wasmer.lib ;\
fi
if [ -f target/release/libwasmer.dylib ]; then \
	cp target/release/libwasmer.dylib package/lib/libwasmer.dylib ;\
fi
if [ -f target/release/libwasmer.so ]; then \
	cp target/release/libwasmer.so package/lib/libwasmer.so ;\
fi
if [ -f target/release/libwasmer.a ]; then \
	cp target/release/libwasmer.a package/lib/libwasmer.a ;\
fi
```
so to be more complete it should be:
```suggestion
	"?package/lib/libwasmer.dll"
	"?package/lib/libwasmer.dll.lib"
	"?package/lib/libwasmer.dylib"
	"?package/lib/libwasmer.so"
	"?package/lib/libwasmer.a"
```

Reply:
> It should be, but this file is OS-independent and Dune **only** supports the `.so` extension on MacOS. So I need to rename the library after building it.

really? It should normally. It's only `.so` on all platforms for OCaml libraries but the system libraries it stays with their builtin names AFAIK

Reply:
> so to be more complete it should [include the `.dll`, `.dll.lib`, `.dylib`, `.so` and `.a` libraries]

That's true, but I only need the `.so` (or the `.dll` on Windows probably).

> It's only .so on all platforms for OCaml libraries but the system libraries stay with their builtin names AFAIK

That is probably correct if it can find it (I think it links using the `-l` linker option), but I force the linker to use the *exact library* in the `~/.opam` directory (which may or may not be the best way to do it, but it is the only way I managed to make it work on Linux and Mac at the same time). Also, `libwasmer` is an OCaml library in this context since it is built and installed by `opam`, inside the `~/.opam` directory.

Reply:
ah i see you've used the OCaml extension manually https://github.com/wasmerio/wasmer-ocaml/blob/d3b2b5721a4013fa4ba942097c3435418c13f095/lib/dune#L5

Mmh, there is an issue though. I've tried to use your library locally to see if it worked and i get this error in the OCaml toplevel:
```
# #use_output "dune top";;
Exception:
Dl.DL_error
 "/home/kit_ty_kate/.opam/default/lib/stublibs/dllctypes-foreign_stubs.so: undefined symbol: wat2wasm".
```
This is because the dynamic library is not present in the default LD_LIBRARY_PATH. So either your users will have to set it manually or I would rather suggest to use static libraries instead even if it makes the resulting binary bigger.

Reply:
I didn't think of the toplevel. However, I did fix the issue with the executables by setting the `rpath` as well, which is added in the linking step for executables (but not the `wasmer` library itself as it is a `-cclib` option). I'm not really sure how to fix the toplevel though...

Reply:
ah i see. If it's only the toplevel then it's fine.